### PR TITLE
Link join steps in onboarding flow

### LINKED ISSUE #665: [Onboarding flow]- Link and create join steps after signup for developers
**Description**
Need to link the existing join step in new onboarding flow and modify according to new flow

### PR DESCRIPTION
**Pull request by vinit717**
##### Example: Closes #665 

Parent ticket: https://github.com/Real-Dev-Squad/todo-action-items/issues/178

### What is the change?
Added more steps in the signup-stepper file to link the join step in it 

### Is Development Tested?

- [x] Yes
- [ ] No

### Example:
https://github.com/Real-Dev-Squad/website-www/assets/111434418/e08ca340-ed12-45ec-9773-72fd14a4fdc9

### CODE REVIEW

File: app/components/stepper-signup.hbs [hunk 1]
```diff
@@ -1,25 +1,66 @@
...
+        @incrementStep={{this.incrementStep}} 
+      />
+    {{else if (eq this.currentStep 2)}}
+      <SignupSteps::stepTwo 
+        @startHandler={{this.startHandler}} 
+      />
+    {{else if (eq this.currentStep 2)}}
+      <SignupSteps::stepTwo 
+        @startHandler={{this.startHandler}} 
+      />
```
Comment (shubhamsigdar1) [inline app/components/stepper-signup.hbs:23]: Two times ??

Comment (shubhamsigdar1) [inline app/components/stepper-signup.hbs:23]: please at step-2 button on props @onClick add @startHandler
image

**Comment by shubhamsigdar1** [comment]
Why Join our discord is seen on button when user choose role developer??

**Comment by vinit717** [comment]
> Why Join our discord is seen on button when user choose role developer?? in video

I have updated the video please have look

**Review by shubhamsigdar1 — APPROVED** [review]

**Review by Ajeyakrishna-k — APPROVED** [review]

File: app/components/stepper-signup.hbs [hunk 2]
```diff
@@ -1,25 +1,62 @@
...
-        />
-      {{else if (eq this.currentStep 2)}}
-        <SignupSteps::stepTwo @letsGoHandler={{this.letsGoHandler}} />
-      {{/if}}
     {{/if}}
   </form>
-</OnboardingCard>
\ No newline at end of file
+
+  {{#if (not (or (eq this.currentStep 0) (eq this.currentStep 1) (eq this.currentStep 2)))}}
```
Comment (satyam73) [inline app/components/stepper-signup.hbs:51]: Can it be simplified like this? I think we're only checking here if step is 2, right?
```suggestion
  {{#if (eq this.currentStep 2)))}}

```

File: app/components/stepper-signup.js [hunk 3]
```diff
@@ -16,15 +20,23 @@ export default class StepperSignupComponent extends Component {
...
+    if (this.currentStep > 0) {
+      this.currentStep -= 1;
+      const queryParams = { dev: true, step: this.currentStep };
+      this.router.transitionTo('join', { queryParams });
+    }
+  }
+
   @action incrementStep() {
-    if (this.currentStep < 5) {
+    if (this.currentStep < 10) {
```
Comment (satyam73) [inline app/components/stepper-signup.js:32]: Can we have this 10 in constant?
MAX_STEP = 10

File: app/components/stepper-signup.js [hunk 4]
```diff
@@ -16,15 +20,23 @@ export default class StepperSignupComponent extends Component {
     );
   }
 
+  @action decrementStep() {
+    if (this.currentStep > 0) {
```
Comment (satyam73) [inline app/components/stepper-signup.js:24]: SImilarly here
MIN_STEP = 10

File: app/components/stepper-signup.hbs (hunk 2 above)
Comment (vinit717) [reply app/components/stepper-signup.hbs:51]: here we are checking that the current step should not 0,1and2 so we can't simplified like this

File: app/components/stepper-signup.js (hunk 3 above)
Comment (vinit717) [reply app/components/stepper-signup.js]: done

File: app/components/stepper-signup.hbs [hunk 5]
```diff
@@ -1,25 +1,62 @@
...
-        alt='RDS-Logo'
-        class='signup-details__rds-logo'
-        data-test-rds-logo
+    {{else if (eq this.currentStep 1)}}
+      <SignupSteps::StepOne 
+        @currentStep={{this.currentStep}} 
+        @incrementStep={{this.incrementStep}} 
+      />
+    {{else if (eq this.currentStep 2)}}
+      <SignupSteps::stepTwo
```
[the remark this answers is not in the document]
Comment (vinit717) [reply app/components/stepper-signup.hbs]: done

**Review by shubhamsigdar1 — APPROVED** [review]

**Review by j24m — APPROVED** [review]

**Pull request merged**